README.md

```diff
@@ -36,3 +36,157 @@ This is the shared library defining public classes that constitute the worker in
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+    // e.cancel       (writable)  (default: false)
+}
+```
+Output of this function would be the boolean value of event's cancel flag.
+This flag is used to determine if that individual document should be processed by the worker.
+
+#### <a name="onProcessDocument"></a>onProcessDocument
+
+This function will be called to process a document. 
+This function is passed `DocumentEventObject` with `Document` as an argument.
+The `onBeforeProcessDocument`  event has already been triggered to check for cancellation flag.
+
+An example of the `DocumentEventObject` is below.
+```
+DocumentEventObject(Document document){
+    document = document;
+}
+```
+For more details  of the <a name="onProcessDocument"></a>`DocumentEventObject`, refer to the java implementation for the class,  [DocumentEventObject](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/DocumentEventObject.java)
+```
+function onProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+This function calls the customization scripts, and if none of them have set the cancellation flag then calls the implementation's `DocumentWorker#processDocument processDocument()` function. 
+
+#### onAfterProcessDocument
+
+This function will be called once the processing of the document completed successfully.
+This function is passed `DocumentEventObject` with `Document` as an argument.
+
+An example of the `DocumentEventObject` is explained in `onProcessDocument` section.
+
+For more details  of the event object,  refer the [onProcessDocument](#onProcessDocument) section. 
+
+```
+function onAfterProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+#### onAfterProcessTask
+
+This is the last function called by worker on the task message.
+This function is passed `TaskEventObject` with `Task` as an argument.
+
+An example of `TaskEventObject` explained in `onProcessTask` section. 
+
+The variables will be initialized with values from task message and it will be sent to the series of other functions in the workflow for further processing.
+
+For more details of the `TaskEventObject`, refer the [onProcessTask](#onProcessTask) section.
+
+```
+function onAfterProcessTask(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // onProcessDocument(e.rootDocument)
+}
+```
+#### onError
+
+This function should be called in case of a failure in the worker that is not handled by the worker code. In chained workers, this will allow continuing to process the document.
```

Comment:
"This function should be called in case of a failure"

Change should to will

Reply:
Done